File: java/LinkedList/insert.java
import java.util.LinkedList;
import java.util.Scanner;

public class LinkedListInsert {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        LinkedList<Integer> list = new LinkedList<>();

        System.out.print("Enter number of elements: ");
        int n = sc.nextInt();

        System.out.println("Enter elements:");
        for (int i = 0; i < n; i++) list.add(sc.nextInt());

        System.out.println("Original LinkedList: " + list);

        // Insert at beginning
        System.out.print("Enter element to insert at beginning: ");
        list.addFirst(sc.nextInt());

        // Insert at end
        System.out.print("Enter element to insert at end: ");
        list.addLast(sc.nextInt());

        // Insert at specific position
        System.out.print("Enter element and position to insert (0-based): ");
        int val = sc.nextInt();
        int pos = sc.nextInt();
        if (pos >= 0 && pos <= list.size()) list.add(pos, val);

        System.out.println("LinkedList after insertions: " + list);
    }
}

File: python/LinkedList/insert.py
class LinkedList(LinkedList):
    def add_first(self, data):
        new_node = Node(data)
        new_node.next = self.head
        self.head = new_node

    def add_at_position(self, data, pos):
        if pos == 0:
            self.add_first(data)
            return
        new_node = Node(data)
        curr = self.head
        for _ in range(pos-1):
            if curr is None:
                print("Position out of bounds")
                return
            curr = curr.next
        new_node.next = curr.next
        curr.next = new_node

# Driver
ll = LinkedList()
n = int(input("Enter number of elements: "))
print("Enter elements:")
for _ in range(n):
    ll.append(int(input()))

ll.add_first(int(input("Enter element to insert at beginning: ")))
ll.append(int(input("Enter element to insert at end: ")))

pos = int(input("Enter position to insert at (0-based): "))
val = int(input("Enter value to insert: "))
ll.add_at_position(val, pos)

print("LinkedList after insertions:")
ll.display()
